core/src/main/java/de/danoeh/antennapod/core/sync/SyncService.java

```diff
@@ -212,12 +182,7 @@ public static void syncImmediately(Context context) {
 
     public static void fullSync(Context context) {
         executeLockedAsync(() -> {
-            context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE).edit()
-                    .putLong(PREF_LAST_SUBSCRIPTION_SYNC_TIMESTAMP, 0)
-                    .putLong(PREF_LAST_EPISODE_ACTIONS_SYNC_TIMESTAMP, 0)
-                    .putLong(PREF_LAST_SYNC_ATTEMPT_TIMESTAMP, 0)
-                    .apply();
-
+            SynchronizationSettings.resetTimestamps();
```

Comment:
Now that it is a dedicated class, I think locking should be done inside that class. Otherwise, future developers could forget that they need to lock and therefore directly call the dedicated class instead of the `SyncService.xy` interface.

Reply:
I can't move the locking inside SynchronizationSettings since i would then drag the whole EventBus and Workmanager handling inside where it does not belong.
Why do i need to lock resetting the timestamps?